File: pinyougou_interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import cn.itcast.core.pojo.good.Brand;
import entity.PageResult;
import entity.SearchEntity;

import java.util.List;
import java.util.Map;

public interface BrandService {
    List<Brand> findAll();


    PageResult findPage(Integer page, Integer rows , Brand brand);

    void save(Brand brand);

    Brand findOne(long id);

    void update(Brand brand);

    void delete(long[] ids);

    List<Map> selectOptionList();

    void updateStatus(long[] ids, String status);
}

File: pinyougou_web_manager/src/main/webapp/js/service/brandService.js
// 定义服务层:
app.service("brandService",function($http){
	this.findAll = function(){
		return $http.get("../brand/findAll.do");
	}
	
	this.findPage = function(page,rows){
		return $http.get("../brand/findPage.do?pageNum="+page+"&pageSize="+rows);
	}

	this.add = function(entity){
		return $http.post("../brand/save.do",entity);
	}
	
	this.update=function(entity){
		return $http.post("../brand/update.do",entity);
	}
	
	this.findOne=function(id){
		return $http.get("../brand/findOne.do?id="+id);
	}
	
	this.dele = function(ids){
		return $http.get("../brand/delete.do?ids="+ids);
	}
	
	this.search = function(page,rows,searchEntity){
		return $http.post("../brand/search.do?page="+page+"&rows="+rows,searchEntity);
	}
	
	this.selectOptionList = function(){
		return $http.get("../brand/selectOptionList.do");
	}
	//审核
    this.updateStatus = function(ids,status){
        return $http.get('../brand/updateStatus.do?ids='+ids+"&status="+status);
    }
    this.AddExcel = function(){
        return $http.get('../brand/AddExcel.do');
    }
    this.derive = function(){

        return $http.post("../brand/derive.do");
    }
});